### PR TITLE
feat: adding typography to tailwind preset

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## **Description**

This pull request adds typography classes to the `design-system-tailwind-preset` package, generating these classes based on the `@metamask/design-tokens` CSS variables. The goal is to standardize typography across Tailwind projects within the MetaMask organization, ensuring consistent design and ease of use in various projects.

This change leverages the design tokens to define typography-related properties such as font sizes, weights, line heights, and letter spacing, allowing developers to apply consistent styles without needing to manually configure these values in each project.

### Key enhancements
- Updates the `@metamask/design-system-tailwind-preset` package to generate tailwind classnames based on the `@metamask/design-tokens`
- Adds Euclid fonts to `storybook-react`
- Adds temporary `Text` stories to `storybook-react` for easier testing and review

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-design-system/issues/56

## **Manual testing steps**

1. Pull this branch
2. Run `yarn build && yarn` to update the preset in `storybook-react` project
3. Apply the typography classes to the storybook typography story
4. Verify that the styles correspond to the design tokens

### **After**

https://github.com/user-attachments/assets/8aefb85c-4a23-47f6-8eac-163abc4b15af

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs).
- [x] I've completed the PR template to the best of my ability.
- [x] I've included tests where applicable.
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable.
- [x] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g., pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and/or screenshots.

### CODE REVIEW

File: apps/storybook-react/stories/Button.stories.ts [hunk 1]
```diff
@@ -4,12 +4,12 @@ import { fn } from '@storybook/test';
...
 
 // More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
-const meta = {
+const meta: Meta<typeof Button> = {
   title: 'Components/Button',
   component: Button,
   tags: ['autodocs'],
   args: { onClick: fn() },
-} satisfies Meta<typeof Button>;
+};
```
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Button.stories.ts:12]: Fixes typescript issue

### Before

Screenshot 2024-10-28 at 4 47 51 PM

### After
Screenshot 2024-10-28 at 4 48 17 PM

File: apps/storybook-react/stories/Text.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Text.stories.tsx:1]: This is a temporary example story to easily review and test the tailwind-preset updates for typography

File: apps/storybook-react/stories/Text.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
```
Comment (georgewrmarshall) [inline apps/storybook-react/stories/Text.tsx:1]: Temporary text component to show how one might use the tailwind preset classnames in a `Text` component can be removed once we have the actual `Text` component

File: apps/storybook-react/tailwind.config.js [hunk 4]
```diff
@@ -8,4 +8,11 @@ module.exports = {
     './src/**/*.{js,jsx,ts,tsx}',
     './stories/**/*.{js,jsx,ts,tsx}',
   ],
+  theme: {
+    extend: {
+      fontSize: {
+        'l-display-md': 'var(--typography-l-display-md-font-size)',
+      },
+    },
+  },
```
Comment (georgewrmarshall) [inline apps/storybook-react/tailwind.config.js:17]: Remove this
```suggestion
```

File: apps/storybook-react/tailwind.css [hunk 5]
```diff
@@ -1,3 +1,38 @@
 @tailwind base;
 @tailwind components;
 @tailwind utilities;
+
```
Comment (georgewrmarshall) [inline apps/storybook-react/tailwind.css:4]: Adding Euclid font-face CSS so fonts display correctly

File: packages/design-system-tailwind-preset/src/index.test.ts [hunk 6]
```diff
@@ -72,4 +72,33 @@ describe('Tailwind Preset', () => {
       }),
     );
   });
+
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/index.test.ts:75]: Adding tests to ensure typography preset objects are present

File: packages/design-system-tailwind-preset/src/index.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 import type { Config } from 'tailwindcss';
 
 import { colors } from './colors';
+import { typography } from './typography';
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/index.ts:4]: Adding typography to main preset configuration file

File: packages/design-system-tailwind-preset/src/typography.test.ts [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    '--typography-s-heading-sm-regular-font-family',
+    '--typography-s-heading-sm-regular-font-size',
+    '--typography-s-heading-sm-regular-line-height',
+    '--typography-s-heading-sm-regular-font-weight',
+    '--typography-s-heading-sm-regular-letter-spacing',
+    '--typography-l-heading-sm-regular-font-family',
+    '--typography-l-heading-sm-regular-font-size',
+    '--typography-l-heading-sm-regular-line-height',
+    '--typography-l-heading-sm-regular-font-weight',
+    '--typography-l-heading-sm-regular-letter-spacing',
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/typography.test.ts:22]: Ignoring deprecated CSS variables

File: packages/design-system-tailwind-preset/src/typography.ts [hunk 9]
```diff
@@ -0,0 +1,164 @@
+export const typography = {
+  fontSize: {
```
Comment (georgewrmarshall) [inline packages/design-system-tailwind-preset/src/typography.ts:2]: These are the current equivalent of what we would consider component tokens. These could be simplified in future with the Typography Figma variant work which we would do at the component level. There are some font weight changes at the variant level which is easiest to define at each variable.

File: apps/storybook-react/tailwind.config.js (hunk 4 above)
Comment (georgewrmarshall) [reply apps/storybook-react/tailwind.config.js]: Updated https://github.com/MetaMask/metamask-design-system/pull/64/commits/8b1e66a35185f0a89e3537ed46a4230e0287e493

**Review by garrettbear — APPROVED** [review]
LGTM

**Review by brianacnguyen — APPROVED** [review]

**Comment by georgewrmarshall** [comment]
@metamaskbot publish-preview

**Pull request merged**